onedal/common/tests/test_sycl.py

```diff
@@ -0,0 +1,118 @@
+
+    raised_exception_dpctl = False
+    raised_exception_backend = False
+
+    try:
+        dpctl_queue = SyclQueue(device)
+    except SyclQueueCreationError:
+        raised_exception_dpctl = True
+
+    try:
+        onedal_queue = onedal_SyclQueue(device)
+    except RuntimeError:
+        raised_exception_backend = True
+
+    assert raised_exception_dpctl == raised_exception_backend
+    # get_device_id must be modified to follow DPCtl conventions
+    # this causes filter_string mismatches
+    # if not raised_exception_backend:
+    #    assert (
+    #        onedal_queue.sycl_device.filter_string
+    #        in dpctl_queue.sycl_device.filter_string
+    #    )
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend or not dpctl_available, reason="requires dpc backend and dpctl"
+)
+@pytest.mark.parametrize("queue", get_queues())
+def test_sycl_queue_conversion(queue):
+    if queue is None:
+        pytest.skip("Not a DPCtl queue")
+    SyclQueue = queue.__class__
+    onedal_SyclQueue = _backend.SyclQueue
+    # convert back and forth to test `_get_capsule` attribute
+
+    q = onedal_SyclQueue(queue)
+    # get_device_id must be modified to follow DPCtl conventions
+    # assert q.sycl_device.filter_string in queue.sycl_device.filter_string
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend or not dpctl_available, reason="requires dpc backend and dpctl"
+)
+@pytest.mark.parametrize("queue", get_queues())
+def test_sycl_device_attributes(queue):
+    from dpctl import SyclQueue
+
+    if queue is None:
+        pytest.skip("Not a DPCtl queue")
+    onedal_SyclQueue = _backend.SyclQueue
+
+    onedal_queue = onedal_SyclQueue(queue)
+
+    # check fp64 support
+    assert onedal_queue.sycl_device.has_aspect_fp64 == queue.sycl_device.has_aspect_fp64
+    # check fp16 support
+    assert onedal_queue.sycl_device.has_aspect_fp16 == queue.sycl_device.has_aspect_fp16
+    # check is_cpu
+    assert onedal_queue.sycl_device.is_cpu == queue.sycl_device.is_cpu
+    # check is_gpu
+    assert onedal_queue.sycl_device.is_gpu == queue.sycl_device.is_gpu
+    # check device number
+    # get_device_id must be modified to follow DPCtl conventions
+    # assert onedal_queue.sycl_device.filter_string in queue.sycl_device.filter_string
+
+
+@pytest.mark.skipif(not _is_dpc_backend, reason="requires dpc backend")
+def test_backend_queue():
+    q = _backend.SyclQueue("cpu")
+    # verify copying via a py capsule object is functional
+    q2 = _backend.SyclQueue(q._get_capsule())
+    # verify copying via the _get_capsule attribute
+    q3 = _backend.SyclQueue(q)
+
+    q_array = [q, q2, q3]
+
+    assert all([queue.sycl_device.has_aspect_fp64 for queue in q_array])
+    assert all([queue.sycl_device.has_aspect_fp16 for queue in q_array])
+    assert all([queue.sycl_device.is_cpu for queue in q_array])
+    assert all([not queue.sycl_device.is_gpu for queue in q_array])
+    assert all(["cpu" in queue.sycl_device.filter_string for queue in q_array])
```

Comment:
Have you manually confirmed that results are as expected with Dummy queue?

Reply:
yep! the only thing off is the device_filter and i have a ticket to fix that as it is older and separate functionality